protocol/dataunit/client_server_test.go

```diff
@@ -56,33 +81,43 @@ func TestClientContextDeadline(t *testing.T) {
 	clientConn, serverConn := net.Pipe()
 	c := &Client{Conn: clientConn}
 	s := &Server{Conn: serverConn}
-	go echoServer(t, ctx, s)
+	echoServerErr := make(chan error)
+
+	go echoServer(ctx, s, echoServerErr)
 
 	wantErr := errors.New("test deadline exceeded")
 	ctx, cancel2 := context.WithDeadlineCause(ctx, time.Now(), wantErr)
 	defer cancel2()
 
 	_, err := c.ExchangeDataUnit(ctx, []byte("hello"))
-	if err != wantErr {
-		t.Errorf("ExchangeDataUnit(): err == %v, expected %v", err, wantErr)
-		t.Fail()
+	if !errors.Is(err, wantErr) {
```

Comment:
I have a feeling @ydnar  was writing this with TinyGo in mind and `errors.Is` is one of its main missing features these days.

Reply:
Ah! OK fair enough I'll undo those changes 👍🏻